File: Introduccion a POO/Problema2/Empleado.java
//2. Crea una clase Empleado con nombre y sueldo
public class Empleado {
    private String nombre;
    private float sueldo;

    public Empleado(String nombre, float sueldo) {
        this.nombre = nombre;
        this.sueldo = sueldo;
    }

    // a) Agrega un método para calcular el sueldo anual
    public void Calcular_Sueldo() {
        System.out.println("El sueldo mensual de: " + sueldo + " , anualmente es de: " + sueldo * 12);
    }

    // b) Agrega un método para aplicar un aumento del 10%
    public void Aplicar_Aumento() {
        float incremento = sueldo + (sueldo * (10f / 100f));
        System.out.println(
                "El sueldo mensual incrementado al 10% es de: " + incremento + " , anualmente es de: "
                        + incremento * 12);
    }

    public static void main(String[] args) {
        System.out.println("Programa Java");
        // c) Crea dos empleados y muestra sus sueldos antes y después del aumento
        Empleado empleado1 = new Empleado("Erick", 1500.99f);
        Empleado empleado2 = new Empleado("Andres", 1800.99f);

        System.out.println("Antes del aumento");
        empleado1.Calcular_Sueldo();
        empleado2.Calcular_Sueldo();
        System.out.println("Despues del aumento");
        empleado1.Aplicar_Aumento();
        empleado2.Aplicar_Aumento();
    }
}

File: Introduccion a POO/Problema2/Empleado.py
# 2. Crea una clase Empleado con nombre y sueldo
class Empleado:
    def __init__(self,nombre, sueldo):
        self.nombre=nombre
        self.sueldo = sueldo

    #a) Agrega un método para calcular el sueldo anual
    def Calcular_Sueldo(self):
        return f"El sueldo mensual de: {self.sueldo} anualmente es de: {self.sueldo*12}"
    
    #b) Agrega un método para aplicar un aumento del 10%
    def Aplicar_Aumento(self):
        incremento= self.sueldo+(self.sueldo*(10/100))
        return f"El sueldo mensual incrementado al 10% es de: {incremento} , anualmente es de: {incremento*12}"

print("Programa python")
#c) Crea dos empleados y muestra sus sueldos antes y después del aumento    
empleado1 = Empleado("David", 1700.75)
empleado2 = Empleado("Hugo", 1900.89)

print("Antes del aumento")
print(empleado1.Calcular_Sueldo())
print(empleado2.Calcular_Sueldo())
print("Despues del aumento")
print(empleado1.Aplicar_Aumento())
print(empleado1.Aplicar_Aumento())

